gettsim/interface.py

```diff
@@ -632,12 +696,23 @@ def _reorder_columns(results):
     return results[sorted_ids + remaining_columns]
 
 
-def _fail_if_root_nodes_are_missing(dag, data):
-    missing_nodes = []
-    for node in _root_nodes(dag):
-        if node not in data and "function" not in dag.nodes[node]:
-            missing_nodes.append(node)
+def _fail_if_root_nodes_are_missing(dag, data, functions):
+    root_nodes = set(_root_nodes(dag))
+
+    # Identify functions that are part of the DAG, but do not depend
+    # on any other function
+    funcs_based_on_params_only = [
```

Comment:
Can you think of a simpler way to do this check, @janosg ?

Reply:
I think the the arguments of the concatenated function would already exclude the functions that only depend on params. So you could skip the construction of `funcs_based_on_params_only`. If you still partial in params before you build the dag, this would be simpler. Otherwise you have to filter out the `params` related arguments similar to line 708. 

But I don't find the current solution bad either.

Reply:
Thanks!

We use `funcs_based_on_params_only` only within this function so the concatenated function does not play a role, The function checks whether all root nodes are in the input data. Since we partial in params before building the DAG, functions which only have params as argument are recognized as root nodes.

So, I don't see a better solution (but am of course open for suggestions)